apps/test/unit/code-studio/pd/form_components_func/FormControllerTest.js

```diff
@@ -52,16 +59,22 @@ describe('FormController', () => {
         options: {},
         requiredFields: [],
         pageComponents: [DummyPage1, DummyPage2, DummyPage3],
-        onSuccessfulSubmit
+        onSuccessfulSubmit,
+        allowPartialSaving: true
       };
     });
 
+    const saveButtonText = 'Save and Return Later';
     const getCurrentPage = () => form.findOne('Pagination').props.activePage;
     const getData = page => form.findOne(page).props.data;
     const getErrors = page => form.findOne(page).props.errors;
     const setPage = i => {
       form.findOne('Pagination').props.onSelect(i + 1);
     };
+    experiments.setEnabled(
+      experiments.TEACHER_APPLICATION_SAVING_REOPENING,
+      true
+    );
```

Comment:
looking at other examples of `setEnabled`, it seems that great care is being taken to only enable the experiment for the duration of each test case, by enabling it during beforeEach and disabling it during afterEach. I believe the reason for this is that an experiment which is set but never cleared might leak over into the rest of the test run. this can lead to flaky/unreproducible test failures, which are the worst. so, even if you are pretty sure this won't affect other tests and/or will launch the experiment soon, I would still suggest you follow this best practice.

TBH I had forgotten about this when I first reviewed -- one of the advantages of checking the experiment in new.js and passing it down is that your tests only have to deal with props, and not experiments. either approach is fine though.

Reply:
Yes, that's great. I moved all of this testing out of FormController, as it's really TeacherApplication that should be asking if it's enabled. I just pushed the changes––I have it in TeacherApplication as a const outside the component, but can change it to be in new.js if you think that's better

Reply:
haha, sorry for splitting this discussion across two threads! I think it is important to be able to test both states, with and without experiment enabled. if you can test it as you have it, then this way is fine. if not, then either useEffect or new.js are fine since we know testing works in both those scenarios.